### PR TITLE
[JENKINS-72474] Additional informations on BFA scans

### PR DESCRIPTION
**Pull request by jgreffe**
https://issues.jenkins.io/browse/JENKINS-72474

Additional informations on `ScanLogAction` for plugins to know the current scan status:
- startTime
- endTime
- exceptionMessage

### Testing done

```[tasklist]
### Submitter checklist
- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue
```

### CODE REVIEW

File: src/main/java/com/sonyericsson/jenkins/plugins/bfa/ScanLogAction.java [hunk 1]
```diff
@@ -64,6 +73,47 @@ public String getLogText() throws IOException {
...
+     */
+    @Exported
+    public Long getEndTime() {
+        return endTime;
+    }
+
+    /**
+     * To call when the scan is finished.
+     */
+    public void finished() {
```
Comment (dwnusbaum) [inline src/main/java/com/sonyericsson/jenkins/plugins/bfa/ScanLogAction.java:97]: I would maybe make this package-private if possible or if not, add `@Restricted(NoExternalUse.class)`.

File: src/main/java/com/sonyericsson/jenkins/plugins/bfa/ScanLogAction.java [hunk 2]
```diff
@@ -20,6 +23,12 @@ public class ScanLogAction implements RunAction2 {
 
     private transient Run run;
 
+    private long startTime = System.currentTimeMillis();
+
+    private Long endTime;
+
+    private Exception exception;
```
Comment (dwnusbaum) [inline src/main/java/com/sonyericsson/jenkins/plugins/bfa/ScanLogAction.java:30]: FWIW, exceptions are not always lightweight from a serialization standpoint. Maybe safer to just store `Exception.toString` as a `String exceptionMessage` or even `boolean scanFailed` and let people look at the log for details.

Comment (jgreffe) [reply src/main/java/com/sonyericsson/jenkins/plugins/bfa/ScanLogAction.java]: See https://github.com/jenkinsci/build-failure-analyzer-plugin/pull/175/commits/aa846cfd31d12d4a4743aab405af584504ac769c

File: src/main/java/com/sonyericsson/jenkins/plugins/bfa/ScanLogAction.java (hunk 1 above)
Comment (jgreffe) [reply src/main/java/com/sonyericsson/jenkins/plugins/bfa/ScanLogAction.java]: See https://github.com/jenkinsci/build-failure-analyzer-plugin/pull/175/commits/aa846cfd31d12d4a4743aab405af584504ac769c

**Review by rsandell — APPROVED** [review]

**Pull request merged**